packages/just-repo-utils/src/readConfigs.ts

```diff
@@ -0,0 +1,89 @@
+import path from 'path';
+import fse from 'fs-extra';
+import { PackageJson, ConfigLoader } from './interfaces/configTypes';
+import { logger } from 'just-task-logger';
+
+type LoaderFn<T> = (pkgPath: string) => T;
+
+/** default JSON load implementation */
+export function loadJson<T = object>(pkgPath: string): T {
+  try {
+    return fse.readJsonSync(pkgPath);
+  } catch (e) {
+    // this emits a warning for the bad file, then returns null (though that is against the return
+    // signature)  This allows a caller to handle null but will mimic { throws: false } behavior
+    logger.warn(`Invalid ${pkgPath.split(path.sep).pop()} detected.`);
+    return (undefined as unknown) as T;
+  }
+}
+
+/** loader implementation for CJSON */
+export function loadCJson<T = object>(pkgPath: string): T {
+  const jju = require('jju');
+  try {
+    const fileAsString = fse.readFileSync(pkgPath, 'utf8').toString();
+    return jju.parse(fileAsString, { mode: 'cjson' });
+  } catch (e) {
+    // this emits a warning for the bad file, then returns null (though that is against the return
+    // signature)  This allows a caller to handle null but will mimic { throws: false } behavior
+    logger.warn(`Invalid ${pkgPath.split(path.sep).pop()} detected.`);
+    return (undefined as unknown) as T;
+  }
+}
+
+/**
+ * A wrapper around checking if a file exists and executing an action on it if it does.  Otherwise returning
+ * undefined.
+ *
+ * @param folder - folder path to use as the base of the config search
+ * @param name - package name, if undefined or falsy the package is assumed to be included in the folder
+ * @param cb - callback function to execute if the file exists
+ */
+function ifConfig<T>(folder: string, name: string | undefined, cb: LoaderFn<T>): T | undefined {
```

Comment:
This has ALMOST the same signature of node's async callback pattern. If you did that, then you could actually make use of async variants of fse functions. Not sure if that is what you're going for, but maybe interesting to implement.

Reply:
It's an interesting idea, the main issue is that the callback here is responsible for directly returning the payload from the callback synchronously.  It would probably make more sense to have a separate async variant.  I'd revisit that later.